### PR TITLE
fix(upstream): correctly find specified version of requested package

### PR DESCRIPTION
**Pull request by jkuri**

### CODE REVIEW

File: src/api/routes.ts [hunk 1]
```diff
@@ -72,6 +72,9 @@ export function getUser(req: auth.AuthRequest, res: express.Response): void | ex
 
 export function getPackage(req: auth.AuthRequest, res: express.Response): void | express.Response {
   let pkgName: string = req.params.package;
+  let baseUrl = req.protocol + '://' + req.get('host');
+  let config = getConfig();
+  let authFile = getAuth();
 
   if (req.headers.referer && req.headers.referer.split(' ')[0] === 'owner') {
     let auth = getAuth();
```
Comment (Izak88) [inline src/api/routes.ts]: You could use `authFile` here if you moved it outside of if statement...

File: src/api/routes.ts [hunk 2]
```diff
@@ -89,18 +91,34 @@ export function getPackage(req: auth.AuthRequest, res: express.Response): void |
 
       res.status(200).json({ name: pkgName, maintainers: users });
     } else {
-      return res.status(404).json('');
+      let pkgJsonPath = getFilePath(`packages/${pkgName}/package.json`);
+      if (pkgName.indexOf('@') !== -1) {
+        pkgName = pkgName.replace(/^(@.*)(\/)(.*)$/g, '$1%2F$3');
+      }
+
+      if (config.useUpstream) {
```
Comment (Izak88) [inline src/api/routes.ts]: this if is exactly the same as if starts on line 130. I think we could extract that in a function. I don't like code duplication... What do you think?

Comment (jkuri) [reply src/api/routes.ts]: okay, I will make it as function.

Comment (jkuri) [reply src/api/routes.ts]: done.

**Review by Izak88 — APPROVED** [review]

**Review by Izak88 — APPROVED** [review]

**Pull request merged**